### PR TITLE
add new platform for modbus board and drivers for energy meter

### PR DESCRIPTION
**Pull request by KwintenSchram**

### CODE REVIEW

File: stack/apps/modbus_forwarder/AcuRev_1312_RCT.c [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    
+    mmodbus_init(modbus_timeout);
+    mmodbus_set32bitOrder(MModBus_32bitOrder_CDAB);
+    DPRINT("acurev inited");
+}
+
+bool acurev_get_real_energy(int64_t *real_energy)
+{
+    uint32_t data;
+    uint16_t scale1, scale2;
```
Comment (L-I-Am) [inline stack/apps/modbus_forwarder/AcuRev_1312_RCT.c]: remove scale2

File: stack/apps/modbus_forwarder/AcuRev_1312_RCT.c [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    success = mmodbus_readHoldingRegister16i(device_address, Real_Energy_Sunpec_Scale_Factor_register, &scale1);
+    success = mmodbus_readHoldingRegister32i(device_address, Total_Real_Energy_Imported_register, &data) && success;
+    *real_energy = (int64_t)((int32_t)data * pow(10, (int16_t)scale1 + 3)); //convert to watt
+    return success;
+}
+
+bool acurev_get_apparent_energy(int64_t *apparent_energy)
+{
+    uint32_t data;
+    uint16_t scale1, scale2;
```
Comment (L-I-Am) [inline stack/apps/modbus_forwarder/AcuRev_1312_RCT.c]: remove scale 2

File: stack/apps/modbus_forwarder/AcuRev_1312_RCT.c [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    DPRINT("acurev inited");
+}
+
+bool acurev_get_real_energy(int64_t *real_energy)
+{
+    uint32_t data;
+    uint16_t scale1, scale2;
+    bool success;
+    success = mmodbus_readHoldingRegister16i(device_address, Real_Energy_Sunpec_Scale_Factor_register, &scale1);
+    success = mmodbus_readHoldingRegister32i(device_address, Total_Real_Energy_Imported_register, &data) && success;
```
Comment (L-I-Am) [inline stack/apps/modbus_forwarder/AcuRev_1312_RCT.c]: use ```&=```?

File: stack/apps/modbus_forwarder/filesystem/button_file.c [hunk 4]
```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright (c) 2015-2021 University of Antwerp, Aloxy NV, LiQuiBit VOF.
```
Comment (L-I-Am) [inline stack/apps/modbus_forwarder/filesystem/button_file.c:2]: do we want the button file for this device?

File: stack/apps/modbus_forwarder/filesystem/energy_file.c [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+    };
+} energy_config_file_t;
+
+
+
+static void file_modified_callback(uint8_t file_id);
+void energy_file_execute_measurement();
+
+static energy_config_file_t energy_config_file_cached
+    = (energy_config_file_t) { .interval = 5, .enabled = true };
```
Comment (L-I-Am) [inline stack/apps/modbus_forwarder/filesystem/energy_file.c]: default 5 seconds interval?

File: stack/apps/modbus_forwarder/inc/AcuRev_1312_RCT.h [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include "stdbool.h"
+
+
+void acurev_1312_rct_init();
+bool acurev_get_real_energy(int64_t *real_energy);
+bool acurev_get_apparent_energy(int64_t *apparent_energy);
```
Comment (L-I-Am) [inline stack/apps/modbus_forwarder/inc/AcuRev_1312_RCT.h:36]: maybe add a comment about what is meant with "real" and "apparent"?

File: stack/apps/modbus_forwarder/inc/mmodbus.h [hunk 7]
```diff
@@ -0,0 +1,140 @@
+#ifndef _M_MODBUS_H_
+#define _M_MODBUS_H_
+
+
```
Comment (L-I-Am) [inline stack/apps/modbus_forwarder/inc/mmodbus.h:4]: I would prefer to put this somewhere in framework but for now, it's ok

File: stack/apps/modbus_forwarder/inc/AcuRev_1312_RCT.h (hunk 6 above)
Comment (KwintenSchram) [reply stack/apps/modbus_forwarder/inc/AcuRev_1312_RCT.h:36]: What do you mean with what is meant? A definition of real energy and apparent energy? Kinda doesn't seem like the place for it.

File: stack/apps/modbus_forwarder/filesystem/button_file.c (hunk 4 above)
Comment (KwintenSchram) [reply stack/apps/modbus_forwarder/filesystem/button_file.c:2]: Yes handy for testing

**Review by L-I-Am — APPROVED** [review]

**Pull request closed without merging**